app/services/puppet_fact_importer.rb

```diff
@@ -1,9 +1,9 @@
 class PuppetFactImporter < StructuredFactImporter
   def self.authorized_smart_proxy_features
-    'Puppet'
+    'Puppet'.freeze
   end
 
   def fact_name_class
-    ::FactName
+    PuppetFactName
```

Comment:
Facts coming from Discovery are FactNames, but I suppose I can say they come from Puppet... after all it's facter who provides them in the image